src/ButtonAnchor.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+export default class ButtonAnchor extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+  render() {
+    let { href, role, tabIndex, disabled, style } = this.props;
+
+    if (disabled) {
+      tabIndex = -1;
+      style = { pointerEvents: 'none', ...style };
+    }
+
+    return (
+      <a
+        {...this.props}
+        style={style}
+        disabled={null}
+        // we want to make sure there is a href attribute on the node
+        // otherwise, the cursor incorrectly styled (except with role='button')
+        href={href || ''}
+        role={role || 'button'}
+        // http://getbootstrap.com/javascript/#dismiss-on-next-click
+        tabIndex={tabIndex || 0}
+        onClick={this.handleClick}
+      />
+    );
+  }
+
+  handleClick(event) {
+    let { disabled, onClick } = this.props;
+
+    event.preventDefault();
```

Comment:
We're obnoxiously inconsistent here.

`<MenuItem>` prevents default whenever `props.disabled` is set: https://github.com/react-bootstrap/react-bootstrap/blob/master/src/MenuItem.js#L17-L19 (not sure what preventing the default accomplishes when there's no `href`, though)

`<NavItem>` unconditionally prevents the default so long as `onSelect` is set: https://github.com/react-bootstrap/react-bootstrap/blob/master/src/NavItem.js#L73-L74